src/prompt/variable.rs

```diff
@@ -146,27 +441,82 @@ pub struct Json;
 
 impl VariableInput for Json {
     fn type_name(&self) -> &str { "json" }
-    fn parse(&self, input: &str) -> Result<Value, Error> {
-        match serde_json::from_str::<serde_json::Value>(input) {
-            Err(e) if e.classify()  == serde_json::error::Category::Eof
-            => Err(Error::Incomplete),
-            Err(e) => Err(no_pos_err(e)),
-            Ok(_) => Ok(Value::Json(
-                model::Json::new_unchecked(input.into())
-            )),
-        }
+    fn parse<'a>(&self, input: &'a str, _flags: InputFlags) -> ParseResult<'a> {
+        // treat as string, and then validate the strings content
+        context(
+            "json",
+            map_res(
+                double_quoted_str,
+                |v| -> Result<Value, anyhow::Error> {
+                    serde_json::from_str::<serde_json::Value>(&v)?;
+
+                    Ok(Value::Json(model::Json::new_unchecked(v.into())))
+                }
+            )
+        )(input)
     }
 }
 
+#[derive(Debug)]
+pub struct Array {
+    pub element_type: Arc<dyn VariableInput>
+}
+
+impl VariableInput for Array {
+    fn type_name(&self) -> &str { "array" }
+    fn parse<'a>(&self, input: &'a str, _flags: InputFlags) -> ParseResult<'a> {
+        context(
+            "array",
+            map(
+                preceded(
+                    char('['),
+                    cut(
```

Comment:
Why is `cut` needed?

Reply:
Mainly for if one element fails to parse it should be unrecoverable

Reply:
I guess it works without

Reply:
It just makes it error out with `nom::internal::Err::Failure` instead of `nom::internal::Err::Error` but it won't change any of the parent parsers behaviour